projects/lib/src/dynamic-views/zone.component.html

```diff
@@ -1,51 +1,50 @@
 <ng-container
-  *ngTemplateOutlet="isArray ? array : single; context: { $implicit: data, first: 'true' }"
+*ngTemplateOutlet="isArray ? arrayTpl : zoneTpl; context: { $implicit: data, first: 'true' }"
 ></ng-container>
 
 <!-- data is an array -->
-<ng-template #array let-data>
-  <ng-container *ngFor="let d of data; let index = index; let first = first" 
-    [ngTemplateOutlet]="single"
+<ng-template #arrayTpl let-data>
+  <ng-container *ngFor="let d of data; let index = index; let first = first"
+    [ngTemplateOutlet]="zoneTpl"
     [ngTemplateOutletContext]="{ $implicit: data, index, first }">
   </ng-container>
 </ng-template>
 
 <!-- data is a single object -->
-<ng-template #single let-data let-index="index" let-first="first">
-  <div *ngIf="!first || (enabled$ | async) === false"
-    [uib-item]="id"
-    [id]="id"
-    [configurable]="false"
-
-    [templates]="templates"
-    
-    [data]="data"
-    [dataIndex]="index"
-    [conditionsData]="conditionsData"
-    
-    (click)="onItemClicked($event, data, index)"
-  ></div>
-  <div #ref *ngIf="first && (enabled$ | async)"
-  
-    class="uib-zone"
-
-    uib-configurable
-    [uib-disable-if]="index"
-
-    [uib-item]="id"
-    [id]="id"
-    [zone]="id"
-    [templates]="templates"
-
-    [data]="data"
-    [dataIndex]="index"
-    [conditionsData]="conditionsData"
-
-    [dndDropzone]="[id]"
-    [dndHorizontal]="ref.isHorizontal()"
-    [dndDisableDropIf]="index"
-    (dndDrop)="onDndDrop($event)"
-
-    (click)="onItemClicked($event, data, index)"
-  ></div>
+<ng-template #zoneTpl let-data let-index="index" let-first="first">
+
+    <!-- en mode editeur -->
+    <uib-item *ngIf="first && (enabled$ | async)" class="uib-zone"
+      [uib-item]="id"
+      [id]="id"
+      [zone]="id"
+      [templates]="templates"
+      [configurable]="true"
+
+      uib-configurable
+
+      [data]="data"
+      [dataIndex]="index"
+      [conditionsData]="conditionsData"
+
+      [attr.data-zone]="id"
+      >
+    </uib-item>
+
+    <!-- en mode normal -->
+    <uib-item *ngIf="!first || (enabled$ | async) === false"
+      [uib-item]="id"
+      [id]="id"
+      [templates]="templates"
+      [configurable]="false"
+
+      [ngClass]="config.classes"
```

Comment:
Same as my comment above: the class should be set by uib-item, because here `config` is in fact a function calling the config from the template

Reply:
useless code. I've removed it